### DISCUSSION TITLE
邏輯思維VS學習思維，學習思維是另類電腦運作方式

### DISCUSSION THREAD

**Discussion by ALAN30359704** [1101-GS4719-C/Python-Community-GS4719-C#1220] [General]
機器學習發展有一個重要的概念是「學習」，我們要掌握AI科技就必須要掌握機器學習的本質，就是「學習思維」。機器學習到底在學甚麼？為什麼讓機器自動學習有這麼強的力量？透過學習賦予資訊科技什麼新生命？

我常用一句話來說明，AI就是將人類經驗系統化，將人類經驗融入資訊系統中，可以跟人一樣，根據環境變化，改變作業方式。人類終於找到一種有效的讓機器自己學習的方法，不用告訴機器用甚麼方法來解決問題，而是透過大量的資料處理，學會推論、預測、辨識我們想知道的答案，這過程就是學習。
我們來比較一下邏輯思維與學習思維就更能夠了解。

一、邏輯思維建立自動化系統

資訊科技將產業作業流程自動化，可以協助我們將人事、會計、財務、倉管、行銷、銷售、服務流程都自動化，其設計方法是人（如系統分析師、程式設計師）透過瞭解需要處理的資料，觀察資料特性及商務規則後，將這些規則歸納成為一套處理邏輯，設計在程式當中，電腦根據我們提供的資料與程式來執行，就會得到人預期的結果，這種運用邏輯概念來制定程式規則的處理方法，就是邏輯思維。

舉一個庫存管理系統的例子，庫存管理是要將倉庫的各庫存品項做好管理，當有進貨時就對進貨品項做入庫增加庫存量；當有需要生產時就將物料品項出庫做扣減庫存量；當有品項低於預先設定的安全庫存量時就提出警示與列出清單，提醒管理者進行採購，這將庫存作業流程自動化，降低管理人力。這就是運用邏輯程式來處理龐大且異動頻繁資料，稱為自動化系統。

資訊科技發展就是用這樣方式來協助產業作業自動化，而這低於安全庫存的品項應該進多少量？這可能跟季節、市場需求、生產良率…等有相關，每次進貨量會隨這些因素變化，這就需要有經驗的人來決定各種品項的採購量，才能夠讓生產順利。
傳統自動化資訊系統會有很多地方需要由人介入，透過人的經驗做適當的決定，來確保系統順利運作。而在AI世代，可以用AI來決定這個品項採購量嗎？能夠讓一般無經驗的人就可以完成這作業嗎? 我們可以透過學習思維來解決這問題。

二、學習思維建立智慧化系統

過往資訊系統大多以邏輯思維來設計，這會受限於人需要設計處理的邏輯流程，但若遇到如影像辨識、文章處理之類應用，很難描述具體邏輯規則，這就出現自動化資訊系統的限制。我們若能夠提供電腦資料及期待處理的結果，電腦就能夠自動地學習這些資料，然後來推論我們期待的結果，這樣就可以處理人類很難處理的大量且龐雜資料，自動學習找出規則，給我們期待的結果，這就是學習思維，也是讓電腦工作的另類方式。

提供資料及期待結果給電腦學習，這是甚麼意思呢？其實資料與期待結果的對應就是人類的經驗，讓電腦自動學習人類經驗的特徵與規則，來解決人類經驗難以系統化的問題，稱為智慧化系統。
例如我們要用人臉辨識來判斷人的年齡及性別，那我們就搜集大量各種人臉圖片這是資料，然後針對每一張人臉圖片標註年齡及性別這是期待結果，這是用我們經驗來標註，這就是將人的經驗提供給電腦的方法。
透過機器來學習這些對應的人臉圖片與年齡、性別，找出圖片資料特徵，建構預測模型，當您提供一張機器從未見過的人臉圖片，機器就能夠用這預測模型，預測可能的年齡與性別，若這預測正確率能夠達到我們的期待，這就是運用學習思維來讓電腦自動學習，以處理我們希望解決的問題，建構一個智慧化系統。

三、自動化系統插上智慧化系統這翅膀，就是AI智慧應用系統

甚麼是AI智慧應用系統？我們回到前述庫存管理例子來說明，若我們能夠提供過往在各種季節、氣候、市場需求、產量、良率…之庫存品項採購量，透過學習思維方法建構庫存品項採購預測模型，就可以來推估可能的各品項進貨量，將自動化系統中需要有經驗的人介入之處，輔以機器學習建立的預測模型，就可以將人的經驗融入系統中，從提供品項進貨量建議，透過不斷累積資料與學習，達到一個期待的正確率，就可以透過模型自動地進行採購作業。

這就是將自動化系統結合智慧化系統，更多人類經驗透過學習思維融入自動化系統中，形成AI智慧應用系統，創造資訊科技前所未有的能力，快速讓過往很難描述規則的問題得到很好的解決方法，各行業的難題都更容易獲得解決，如服務業的行銷成效預估、自然對話服務機器人；製造業的瑕疵檢測、預修保養；醫療業的病徵檢測、癌症處方推薦；教育業的能提早了解學習成效的評估系統、輔助教學系統；金融業的授信判斷、理財推薦；人資管理的自動履歷篩檢、離職預測…等各行各業的應用方案，協助個人工作也讓產業得以快速發展，賦能個人及賦能產業。

掌握學習思維，對AI有更深認識，也了解AI學習的本質，能夠從您對產業的了解，選對問題，運用學習思維建構智慧化解決方案，讓我們對於機器能力有更完整的應用，賦予智慧應用有效地解決問題。

**Comment by DuxtinTai** [comment]
確實，在AI的發展中，除了機器學習以外，還有許多其他的技術，如深度學習、自然語言處理、影像辨識等等。這些技術可以應用在不同的領域，幫助人們解決各種問題。

然而，機器學習的發展也帶來了一些挑戰和風險。例如，當機器學習模型基於過去的數據進行預測時，有可能會導致偏見和歧視的問題。這是因為數據本身可能存在偏見或歧視，或者模型的設計可能會強化某些族群或人群的偏見。因此，在機器學習的應用中，需要注意避免這些問題的出現，並盡力確保模型的公正性和透明度。

因此，我認為機器學習的發展帶來了許多的機會和挑戰，我們需要深入了解其本質和應用，以更好地應對未來的發展趨勢。
